### PR TITLE
Put bindings crates in the workspace

### LINKED ISSUE #1889: Should bindings be part of the workspace?

### PR DESCRIPTION
**Pull request by thibault-martinez**
Fixes https://github.com/iotaledger/wallet.rs/issues/1889

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -12,10 +12,10 @@ on:
       - dev
```
Comment (kwek20) [inline .github/workflows/build-and-test.yml:12]: lets also remove `dev`

Comment (thibault-martinez) [reply .github/workflows/build-and-test.yml:12]: There are a lot of issues with the workflow files, I'll do a second PR

File: .gitignore [hunk 2]
```diff
@@ -15,7 +15,6 @@ alice-database
 /walletdb
 /pingdb
 /pongdb
-Cargo.lock
 !wallet/bindings/**/Cargo.lock
```
Comment (kwek20) [inline .gitignore]: i dont think we need this ignore anymore now

File: wallet/bindings/python/Cargo.toml [hunk 3]
```diff
@@ -12,6 +10,10 @@ repository = "https://github.com/iotaledger/wallet.rs"
 license = "Apache-2.0"
 keywords = [ "iota", "wallet", "transaction", "python" ]
 categories = [ "cryptography::cryptocurrencies" ]
+publish = false
+
+[package.metadata.maturin]
+name = "iota-wallet"
```
Comment (kwek20) [inline wallet/bindings/python/Cargo.toml]: in iota.rs we just removed it and changed it to `pyproject.toml` (https://github.com/iotaledger/iota.rs/pull/1629/files#diff-6ae29472b05324aec56b7d87be8614015aaf98b1c3ead751da069830381cf8eb)
```
[project]
name = "iota-client"
```

Comment (thibault-martinez) [reply wallet/bindings/python/Cargo.toml]: @Thoralf-M were you still investigating something? You said my PR wasn't working

Comment (Thoralf-M) [reply wallet/bindings/python/Cargo.toml]: iota.rs is fine now, just apply the same changes to wallet.rs

File: wallet/bindings/python/tox.ini [hunk 4]
```diff
@@ -14,5 +14,5 @@ description = Run the unit tests under {basepython}
 passenv = *
 deps = -r requirements-dev.txt
 commands =
-    pip install .
+    python setup.py install
```
Comment (Thoralf-M) [inline wallet/bindings/python/tox.ini]: Also undo this change if you do the other changes

Comment (thibault-martinez) [reply wallet/bindings/python/tox.ini]: Yes sir

**Review by kwek20 — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]
